src/deepsparse/loggers/function_logger.py

```diff
@@ -0,0 +1,111 @@
+# limitations under the License.
+
+"""
+Implementation of the Function Logger
+"""
+
+from typing import Any, Dict, List
+
+from deepsparse.loggers import BaseLogger, MetricCategories
+from deepsparse.loggers.metric_functions import apply_function
+
+
+__all__ = ["FunctionLogger"]
+
+
+class FunctionLogger(BaseLogger):
+    """
+    DeepSparse logger that applies functions to raw log values
+    according to the specified config file
+
+    :param loggers: A DeepSparse Logger object
+    :param config: A configuration dictionary that specifies the mapping between
+        the target name and the functions to be applied to raw log values
+        e.g.
+        ```
+        {"pipeline_inputs":
+            [{"function": "identity",
+              "frequency": 3},
+
+            [{"function": "identity",
+              "frequency": 5}],
+
+         "pipeline_outputs":
+            [{"function": "identity",
+            "frequency": 4}]
+        }
+    """
+
+    def __init__(
+        self,
+        logger: BaseLogger,
+        config: Dict[str, List[Dict[str, Any]]],
+    ):
+
+        self.logger = logger
+        self.config = self._add_frequency_counter(config)
+
+    def log(self, identifier: str, value: Any, category: MetricCategories):
+        """
+        :param identifier: The identifier of the log
+             By default should consist of at least one string (name of the pipeline):
+                e.g. identifier = "pipeline_name"
+            If identifier is to be more complex, optional strings are to be concatenated
+            using dot as a separator:
+                e.g. identifier = "pipeline_name.some_argument_1.some_argument_2"
+        :param value: The data structure that the logger is logging
+        :param category: The metric category that the log belongs to
+        """
+        if category == MetricCategories.DATA:
+            pipeline_name, *target_name = identifier.split(".")
+
+            list_functions = self.config.get(".".join(target_name))
+            if list_functions is None:
+                return
+
+            for function_dict in list_functions:  # iterate over all available functions
+                if function_dict["frequency_counter"] % function_dict["frequency"] == 0:
+                    # if count matches the frequency, log
+                    # otherwise, pass
+                    self._log_data(identifier, value, category, function_dict)
+                # increment the counter
+                function_dict["frequency_counter"] += 1
+
+        else:
+            self.logger.log(identifier=identifier, value=value, category=category)
+
+    def _log_data(
+        self,
+        identifier: str,
+        value: Any,
+        category: MetricCategories,
+        function_dict: Dict[str, Any],
+    ):
+        # fetch the string signature & apply function
+        # specified by the signature to the value
+        function_signature = function_dict.get("function")
```

Comment:
signature is a bit misleading, let's change to `_name` if it's a string

Reply:
It is a string for now, but eventually we would like to map strings into functions on the server level, not logger level. So ultimately, what is now called `function_signature` will be a callable. For now I will just rename it to `function`.